substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -516,12 +524,26 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
 
             JavaKind returnResultKind = b.getInvokeReturnType().getJavaKind().getStackKind();
             ValueNode transplantedSingleFunctionality = null;
+            NewInstanceNode transplantedNewInstance = null;
             if (singleFunctionality instanceof Invoke) {
+
                 Invoke singleInvoke = (Invoke) singleFunctionality;
                 MethodCallTargetNode singleCallTarget = (MethodCallTargetNode) singleInvoke.callTarget();
                 ResolvedJavaMethod resolvedTarget = lookup(singleCallTarget.targetMethod());
 
-                maybeEmitClassInitialization(b, singleCallTarget.invokeKind() == InvokeKind.Static, resolvedTarget.getDeclaringClass());
+                if (singleNewInstance != null) {
+                    ResolvedJavaType type = null;
+                    if (singleNewInstance instanceof DynamicNewInstanceNode) {
+                        type = lookup(originalProviders.getConstantReflection().asJavaType(((DynamicNewInstanceNode) singleNewInstance).getInstanceType().asConstant()));
+                    }
+                    if (singleNewInstance instanceof NewInstanceNode) {
+                        type = lookup(((NewInstanceNode) singleNewInstance).instanceClass());
+                    }
+                    maybeEmitClassInitialization(b, true, resolvedTarget.getDeclaringClass());
+                    transplantedNewInstance = b.add(new SubstrateNewInstanceNode(type, true));
+                } else {
+                    maybeEmitClassInitialization(b, singleCallTarget.invokeKind() == InvokeKind.Static, resolvedTarget.getDeclaringClass());
```

Comment:
Why do you need this class initialization check for the invoke when there is a `singleNewInstance`?

Reply:
I only added this during debugging thinking that it was the cause of a problem I was having. I'll remove it.